File: VB/ViewModel/XPO/InstantFeedbackMode/Issues/Issue.vb
Imports DevExpress.Xpo

Namespace Issues
    Public Class Issue
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
            Created = Date.Now
        End Sub

        Private _Subject As String

        <Size(200)>
        Public Property Subject As String
            Get
                Return _Subject
            End Get
            Set(ByVal value As String)
                SetPropertyValue(NameOf(Issue.Subject), _Subject, value)
            End Set
        End Property

        Private _UserId As Integer

        Public Property UserId As Integer
            Get
                Return _UserId
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(NameOf(Issue.UserId), _UserId, value)
            End Set
        End Property

        Private _User As User

        <Association("UserIssues")>
        Public Property User As User
            Get
                Return _User
            End Get
            Set(ByVal value As User)
                SetPropertyValue(NameOf(Issue.User), _User, value)
            End Set
        End Property

        Private _Created As Date

        Public Property Created As Date
            Get
                Return _Created
            End Get
            Set(ByVal value As Date)
                SetPropertyValue(NameOf(Issue.Created), _Created, value)
            End Set
        End Property

        Private _Votes As Integer

        Public Property Votes As Integer
            Get
                Return _Votes
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(NameOf(Issue.Votes), _Votes, value)
            End Set
        End Property

        Private _Priority As Priority

        Public Property Priority As Priority
            Get
                Return _Priority
            End Get
            Set(ByVal value As Priority)
                SetPropertyValue(NameOf(Issue.Priority), _Priority, value)
            End Set
        End Property
    End Class

    Public Enum Priority
        Low
        BelowNormal
        Normal
        AboveNormal
        High
    End Enum
End Namespace

File: CS/ViewModel/XPO/InfiniteAsyncSource/Issues/Issue.cs
using System;
using DevExpress.Xpo;

namespace XPOIssues.Issues {
    public class Issue : XPObject {
        public Issue(Session session) : base(session) {
            Created = DateTime.Now;
        }
        string _Subject;
        [Size(200)]
        public string Subject {
            get { return _Subject; }
            set { SetPropertyValue(nameof(Subject), ref _Subject, value); }
        }

        int _UserId;
        public int UserId {
            get { return _UserId; }
            set { SetPropertyValue(nameof(UserId), ref _UserId, value); }
        }

        User _User;
        [Association("UserIssues")]
        public User User {
            get { return _User; }
            set { SetPropertyValue(nameof(User), ref _User, value); }
        }

        DateTime _Created;
        public DateTime Created {
            get { return _Created; }
            set { SetPropertyValue(nameof(Created), ref _Created, value); }
        }

        int _Votes;
        public int Votes {
            get { return _Votes; }
            set { SetPropertyValue(nameof(Votes), ref _Votes, value); }
        }

        Priority _Priority;
        public Priority Priority {
            get { return _Priority; }
            set { SetPropertyValue(nameof(Priority), ref _Priority, value); }
        }
    }

    public enum Priority { Low, BelowNormal, Normal, AboveNormal, High }
}
